### DISCUSSION TITLE
How to change all nodenames to lower case?

### DISCUSSION THREAD

**Discussion by Kosiare** [prometheus-operator/kube-prometheus#792] [Q&A] [answered]
Hi everyone!

I've installed via Helm kube-prometheus-stack from prometheus-community repository. Everything works fine but I got a problem with node/host names.

I'm using AKS with one node pool and 8 nodes. An example node hostname is: _aks-nodepool-0000A._ When I use command _kubectl get nodes_ I can see that the node name is a little bit different because it's written in lowercase, _for example aks-nodepool-0000a_.

So that makes problem with metrics because some metrics are using lowercase node name like _kube_node_status_capacity_ 
![kube_node_status_capacity](https://user-images.githubusercontent.com/26820412/99387864-f3006800-28d4-11eb-936b-f20410827250.png)
and other metrics are with uppercase last letter, for example this one: _node_uname_info_.
![node_uname_info](https://user-images.githubusercontent.com/26820412/99387867-f4319500-28d4-11eb-88a8-072087dd628c.png)

This makes really a mess in my Grafana dashboards because I have to use two variables (one with lowercase and one with uppercase name).I tried to use case insensitive expression `kube_node_status_capacity{node=~"(?i:($node))"},resource="cpu"} `but it does not work in any query.

Is there a solution to change all nodename node values to lowercase? AFAIK it's not possible to change it in AKS (I would have to recreate whole cluster what is obviously a no-no for me because it's productive cluster).

**Comment by brancz** [comment]
Just to make sure I understand correctly, the `kube_node_status_capacity` exposes what you expect, but `node_uname_info` does not?

**Reply by Kosiare** [reply]
That's the example (there's a lot of metrics with this issue) but yes - `kube_node_status_capacity` gives me `node` value as a lowercase but `node_uname_info` gives me `nodename` with last character as a uppercase

**Reply by brancz** [reply]
hmm it doesn't appear that the node-exporter does anything to the string https://github.com/prometheus/node_exporter/blob/8b466360a35581e0301bd22918be7011cf4203c3/collector/uname_linux.go#L35

could you run `uname -n` on one of these machines and check what that returns?

From what I can tell the exporters are exporting exactly the data that is handed to them, and the kernel/kubernetes may be doing weird things here. Possibly AKS is doing this under the hood as node naming in kubernetes is technically free form, if this is the case I would open a support ticket with Azure.

**Reply by Kosiare** [reply]
Yeah, I think you're right because `uname -n` gave me output with last letter in uppercase. Is there an easy way to convert all values to lowercase or the last thing I can do is open a support ticket in Azure?

**Reply by brancz** [reply]
I don't believe that's possible today, the closest discussion appears to be https://github.com/prometheus/prometheus/issues/1548

**Reply by Kosiare** [reply]
Okay, so I'll open ticket to Azure. I'll let you know if they know any solution.

**Reply by Kosiare** [reply]
So, basically they cannot do anything and told me to use tags in Azure Portal (which is funny because it won't work for sure - how Prometheus would scrape something from their website...).

**Comment by SuperQ** [comment, marked as answer]
This can now be fixed with the Prometheus `lowercase` relabel action.

https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config
